### ISSUE TITLE
Attachments support is not provided?

### ISSUE THREAD

**Issue by prasadKodeInCloud** [gfk-ba/meteor-mailgun-api#3]
Hi,

Could not send attachments as per the documentation in npm 'mailgun-js' package. https://www.npmjs.org/package/mailgun-js

This feature is not implemented yet?

**Comment by jerefrer** [comment]
Any news on this ? I'd love to be able to choose the name of the attached file ;)

**Comment by marcodejongh** [comment]
@jerefrer  The send method of this packages is just a simple wrapper around mailgun-js taking care of a lot of oddities in the mailgun api.

So the following should work:

```
var filename = 'mailgun_logo.png';
var filepath = path.join(__dirname, filename);
var file = fs.readFileSync(filepath);
 
var attch = new MailgunInstance.api.Attachment({data: file, filename: filename});
 
var data = {
  from: 'Excited User <me@samples.mailgun.org>',
  to: 'serobnic@mail.ru',
  subject: 'Hello',
  text: 'Testing some Mailgun awesomness!',
  attachment: attch
};
 
mailgunInstance.send(data);
```

Although it would be nicer if the filepath and file read are also added to the wrapper I agree. But I haven't needed attachment support myself yet, so thats why I didn't add it.

So I'll change this to a feature and add it if I find the time.

**Comment by jerefrer** [comment]
Haven't time to try this just now but if Attachment is accessible is should work just fine. That's all I needed thank you :)

If you don't have time to implement it properly maybe you could document this on the README for everyone to see ?

**Comment by marcodejongh** [comment]
@jerefrer Done, I've also updated the workaround to use a fiber instead of a sync readFile

**Comment by jerefrer** [comment]
I sure will but it won't be before a week or so. Thanks for the doc, doc :)

**Comment by shkomg** [comment]
it'd be great to have it working but it doesn't work like that.

it even just doesn't works when I supply file data as Buffer object to the attachment property.

Any ideas?

Your help is really needed!

Thank you.

**Comment by marcodejongh** [comment]
@shkomg What is the error you get?

**Comment by shkomg** [comment]
@marcodejongh 

I got no error - email is send but with no attachment.

Here is the code (i'm sending the PDF out);

```
        var toc = new PDFDocument({size: 'A4', margin: 50});
        toc.fontSize(12);
        toc.text('Invoice', 10, 30, {align: 'center', width: 200});
        toc.writeSync(path);

        var readFile = Meteor.wrapAsync(Npm.require('fs').readFile)
        var file = readFile(path);

        Meteor.Mailgun.send({
            to: doc.email,
            from: doc.from,
            subject: 'Invoice for your ad campaign',
            text: message,
            html: Handlebars.templates['contactEmail']({siteURL: Meteor.absoluteUrl(), fromName: doc.name, fromEmail: doc.from, message: message}),
            attachment: file
        });
```

And when I tried previously proposed solution I can't reach ```MailgunInstance.api.Attachment```

**Comment by marcodejongh** [comment]
@shkomg  With MailgunInstance I meant the pointer to the instance of mailgun.

So it looks like you keep a instance of mailgun on ```Meteor.Mailgun```. So in your case you should use ```Meteor.Mailgun.Attachment```.

**Comment by shkomg** [comment]
@marcodejongh 

ok, when I do this ```var attch = new Meteor.Mailgun.Attachment({data: file, filename: "invoice.pdf"}); ```

Meteor throws error:

```Exception while invoking method 'getInvoice' TypeError: undefined is not a function ```

**Comment by marcodejongh** [comment]
@shkomg You have a getInvoice method somewhere?

**Comment by shkomg** [comment]
@marcodejongh 

yes, all this code is in getInvoice method & that error points on this string: ```var attch = new Meteor.Mailgun.Attachment({data: file, filename: "invoice.pdf"}); ```

It happens so that Meteor.Mailgun is an object and it just doesn't have Attachment method.

I also have tried to use just ``` new Mailgun.Attachment``` and ```new Mailgun({api key & domain}).Attachment``` - all these are objects without Attachment

**Comment by marcodejongh** [comment]
@shkomg Try: 

```
var attch = new Meteor.Mailgun.api.Attachment({data: file, filename: "invoice.pdf"});
```

**Comment by shkomg** [comment]
@marcodejongh 

here it is ```TypeError: Cannot read property 'Attachment' of undefined```

**Comment by marcodejongh** [comment]
@shkomg 
This should definitely work:
```new Mailgun({api key & domain}).api.Attachment```

**Comment by shkomg** [comment]
@marcodejongh it works - it send's email but with no attachment :(

**Comment by marcodejongh** [comment]
@shkomg Ok I'll add the support the mailgun this weekend that will make it a lot easier

**Comment by shkomg** [comment]
@marcodejongh that'd be awesome!

Thank you!

**Comment by shkomg** [comment]
@marcodejongh Great, that change in readme works! Thank you.

**Comment by marcodejongh** [comment]
@shkomg  Thank you for testing my readme.md ;)

**Comment by quape** [comment]
I tried to add an attachment with:

    'attachment': new mg.api.Attachment({data: Assets.getBinary('logo.png'), filename: 'logo.png'})

wich did not work. The mail was sent but without attachment.

Secondly, if I can make it work, is it possible to mark the attachment as an inline attachment? It's for inline images with are not shown as attachment files in the mail app.

Thanks
